packages/api/src/controllers/taskController.js

```diff
@@ -167,6 +167,27 @@ const taskController = {
     }
   },
 
+  async patchWage(req, res) {
+    try {
+      const { task_id } = req.params;
+      const { wage_at_moment } = req.body;
+
+      //Ensure only adminRoles can modify task wage
+      if (!adminRoles.includes(req.role)) {
```

Comment:
I think this same thing be accomplished by adding another checkScope('edit:user_wage?') to the taskRoute.

I see there is another patch in this controller above this one that may handling permissions dangerously.

Reply:
Thank you, that's much better.

Do you mean we should add a permission for `patchTaskDate` so we can use `checkScope`?

Reply:
I meant that checkScope() checks and verifies the role right. Since this is editing the user_wage that the existing edit:user_wage permission can be added to the patchwage task route you made.